internal/pkg/deploy/cloudformation/stack/static_site_integration_test.go

```diff
@@ -0,0 +1,103 @@
+//go:build integration || localintegration
+
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package stack_test
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/aws/copilot-cli/internal/pkg/config"
+	"github.com/aws/copilot-cli/internal/pkg/deploy/cloudformation/stack"
+	"github.com/aws/copilot-cli/internal/pkg/manifest"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
+)
+
+func TestStaticSiteService_TemplateAndParamsGeneration(t *testing.T) {
+	const (
+		appName = "my-app"
+	)
+	envName := "my-env"
+
+	testDir := filepath.Join("testdata", "workloads")
+
+	tests := map[string]struct {
+		ManifestPath        string
+		TemplatePath        string
+		ParamsPath          string
+		EnvImportedCertARNs []string
+	}{
+		"simple": {
+			ManifestPath: filepath.Join(testDir, "static-site-manifest.yml"),
+			TemplatePath: filepath.Join(testDir, "static-site.stack.yml"),
+			ParamsPath:   filepath.Join(testDir, "static-site.params.json"),
+		},
+	}
+
+	// run tests
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			// parse files
+			manifestBytes, err := os.ReadFile(tc.ManifestPath)
+			require.NoError(t, err)
+			tmplBytes, err := os.ReadFile(tc.TemplatePath)
+			require.NoError(t, err)
+			paramsBytes, err := os.ReadFile(tc.ParamsPath)
+			require.NoError(t, err)
+
+			dynamicMft, err := manifest.UnmarshalWorkload([]byte(manifestBytes))
+			require.NoError(t, err)
+			require.NoError(t, dynamicMft.Validate())
+			mft := dynamicMft.Manifest()
+
+			envConfig := &manifest.Environment{
+				Workload: manifest.Workload{
+					Name: &envName,
+				},
+			}
+			serializer, err := stack.NewStaticSite(&stack.StaticSiteConfig{
+				App: &config.Application{
+					Name: appName,
+				},
+				EnvManifest: envConfig,
+				Manifest:    mft.(*manifest.StaticSite),
+				RuntimeConfig: stack.RuntimeConfig{
+					EnvVersion: "v1.42.0",
+				},
+				AssetMappingURL: "https://stackset-bucket.s3-us-west-2.amazonaws.com/mappingfile",
```

Comment:
works with https, but we actually pass in `s3://bucket/path` https://github.com/aws/copilot-cli/blob/ea86849093b9ffb03a8efec4cb595ad4fd10fc4a/internal/pkg/cli/deploy/static_site.go#L111